tests/unity/unity_test.cmake

```diff
@@ -0,0 +1,135 @@
+#
+# Copyright (c) 2019 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+include($ENV{ZEPHYR_BASE}/cmake/app/boilerplate.cmake NO_POLICY_SCOPE)
+
+set(UNITY_PRODUCTS_DIR "${APPLICATION_BINARY_DIR}/unity")
+set(CMOCK_PRODUCTS_DIR "${UNITY_PRODUCTS_DIR}/mocks")
+set(CMOCK_DIR "${ZEPHYR_BASE}/../cmock")
+set(RUBY_EXEC "/usr/bin/ruby")
```

Comment:
find_package(Ruby \<whatever the CMock minimimum supported Ruby version is>)

/usr/bin/ruby will then be found in ${RUBY_EXECUTABLE}.

https://github.com/Kitware/CMake/blob/master/Modules/FindRuby.cmake

Reply:
I've tried to add that but it resulted in some errors which are confusing. Errors mixed with `found suitable version "2.5.1", minimum required is "2.0.0"`. Do you have any suggestions?
```
CMake Error: Error required internal CMake variable not set, cmake may not be built correctly.
Missing variable is:
CMAKE_FIND_LIBRARY_PREFIXES
CMake Error: Error required internal CMake variable not set, cmake may not be built correctly.
Missing variable is:
CMAKE_FIND_LIBRARY_SUFFIXES
-- Could NOT find Ruby (missing: RUBY_INCLUDE_DIR RUBY_CONFIG_INCLUDE_DIR) (found suitable version "2.5.1", minimum required is "2.0.0")
```

Reply:
I was not able to figure this out either.

Since we don't need the ruby include dirs yet, I'd suggest that we use the simpler find_program instead.

So

```
find_program(
  RUBY
  ruby
  )
```